src/freenas/usr/local/bin/execute_dump.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+. /usr/local/share/system_info
+
+VMCORE_FILE=/proc/vmcore
+KDUMP_SCRIPT=/usr/sbin/kdump-config
+KDUMP_DEFAULTS=/etc/default/kdump-tools
+[ -r $KDUMP_DEFAULTS ] && . $KDUMP_DEFAULTS
+
+# If no vmcore file exists, system is just booting and we should load the crash kernel
+if [ ! -e "$VMCORE_FILE" -o ! -s "$VMCORE_FILE" ]; then
+  echo "Starting kdump-tools"
+  $KDUMP_SCRIPT load
+  exit 0
+fi
+
+sys_pool=$(${FREENAS_SQLITE_CMD} ${FREENAS_CONFIG} "
+    SELECT
+        sys_pool
+    FROM
+        system_systemdataset
+    ORDER BY
+        -id
+    LIMIT 1
+  ")
+
+if [ -z "$sys_pool"]; then
+  echo "No system dataset pool found. Aborting dump."
+  reboot -f
+fi
+
+zpool list | grep -q "$sys_pool"
+if [ $? -eq 1 ]; then
```

Comment:
You mean -eq 0?

Reply:
No, nevermind :)

Reply:
Nope, if we are not able to find that pool, then we try to import it